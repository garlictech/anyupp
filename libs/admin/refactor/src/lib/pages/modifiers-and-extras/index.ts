export * from './admin-pages-modifiers-and-extras.module';
export { ProductComponentSetFormComponent } from './components/product-component-set-form/product-component-set-form.component';
export { ModifiersAndExtrasFormService } from './services/modifiers-and-extras-form.service';
export { ModifiersAndExtrasListService } from './services/modifiers-and-extras-list.service';
