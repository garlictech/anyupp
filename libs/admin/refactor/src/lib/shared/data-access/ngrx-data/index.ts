export { BaseCollectionService } from './services/base-collection.service';
export * from './const';
