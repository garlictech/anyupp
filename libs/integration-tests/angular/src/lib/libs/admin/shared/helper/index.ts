import { Auth } from 'aws-amplify';

import { awsConfig } from '@bgap/crud-gql/api';
import {
  getCognitoUsername,
  testAdminUserPassword,
} from '@bgap/shared/fixtures';

export const signInToCognito = async () => {
  Auth.configure({
    ...awsConfig,
    authenticationFlowType: 'USER_PASSWORD_AUTH',
    aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
  });

  await Auth.signIn(getCognitoUsername('monad'), testAdminUserPassword);
};

export const signOutFromCognito = async () => {
  await Auth.signOut();
};
