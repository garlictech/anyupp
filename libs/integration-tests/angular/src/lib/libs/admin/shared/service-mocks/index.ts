export class MockNbDialogService {
  open() {
    /** */
  }
}

export class MockTranslateService {
  get() {
    /** */
  }
}
