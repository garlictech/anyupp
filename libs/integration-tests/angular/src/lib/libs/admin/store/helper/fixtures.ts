export const testFixtures = [
  {
    id: 'test_fixture_id_1',
    name: 'Text fixture 1',
  },
  {
    id: 'test_fixture_id_2',
    name: 'Text fixture 2',
  },
  {
    id: 'test_fixture_id_3',
    name: 'Text fixture 3',
  },
];
