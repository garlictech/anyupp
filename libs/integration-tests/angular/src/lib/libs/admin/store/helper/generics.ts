/* eslint-disable @typescript-eslint/no-explicit-any */

import { of, range } from 'rxjs';
import { delay, last, switchMap, take, tap } from 'rxjs/operators';

export const emptyListTestLogic = (service: any, done: jest.DoneCallback) => {
  service.entities$
    .pipe(
      take(1),
      tap(items => {
        // The initial list is empty
        expect(items).toEqual([]);
      }),
    )
    .subscribe(() => {
      done();
    });
};

export const filteredListTestLogic = (
  service: any,
  dataFixtures: any[],
  filter: any,
  done: jest.DoneCallback,
) => {
  const sdkSpy = jest.spyOn(service.apiConf, 'add');

  range(0, dataFixtures.length)
    .pipe(
      tap(idx => {
        sdkSpy.mockImplementation(
          jest.fn().mockReturnValue(of(dataFixtures[idx])),
        );
        service.setFilter(filter);
      }),
      switchMap(idx => service.add$(dataFixtures[idx])),
      last(),
      switchMap(() => service.entities$),
      take(1),
      tap(items => {
        // Full list
        expect(items).toMatchSnapshot();
      }),
      switchMap(() => service.filteredEntities$),
      take(1),
      tap(items => {
        // Filtered list
        expect(items).toMatchSnapshot();
      }),
    )
    .subscribe(() => {
      done();
    });
};

export const addEntityToListTestLogic = (
  service: any,
  dataFixture: any,
  done: jest.DoneCallback,
) => {
  const sdkSpy = jest
    .spyOn(service.apiConf, 'add')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  service
    .add$(dataFixture)
    .pipe(
      tap(() => {
        // "Add" method has been called
        expect((sdkSpy.mock.calls[0][0] as any).input).toMatchSnapshot();
      }),
      switchMap(() => service.entities$),
      take(1),
      tap(items => {
        // The list has been updated
        expect(items).toMatchSnapshot();
      }),
    )
    .subscribe(() => {
      done();
    });
};

export const updateEntityListTestLogic = (
  service: any,
  dataFixture: any,
  updateFixture: any,
  done: jest.DoneCallback,
) => {
  const sdkCreateSpy = jest
    .spyOn(service.apiConf, 'add')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  const sdkUpdateSpy = jest
    .spyOn(service.apiConf, 'update')
    .mockImplementation(jest.fn().mockReturnValue(of(updateFixture)));

  service
    .add$(dataFixture)
    .pipe(
      tap(() => {
        // "Add" method has been called
        expect((sdkCreateSpy.mock.calls[0][0] as any).input).toMatchSnapshot();
      }),
      switchMap(() => service.update$(updateFixture)),
      tap(() => {
        // "Update" method has been called
        expect((sdkUpdateSpy.mock.calls[0][0] as any).input).toMatchSnapshot();
      }),
      switchMap(() => service.entities$),
      take(1),
      tap(items => {
        // The list has been updated
        expect(items).toMatchSnapshot();
      }),
    )
    .subscribe(() => {
      done();
    });
};

export const patchFilterTestLogic = (
  service: any,
  initialFilter: Record<string, unknown>,
  newFilter: Record<string, unknown>,
  done: jest.DoneCallback,
) => {
  of('test')
    .pipe(
      tap(() => {
        // Initial filter
        service.setFilter(initialFilter);
      }),
      switchMap(() => service.filter$),
      take(1),
      tap(filters => {
        // Filter has been updated
        expect(filters).toMatchSnapshot();
        service.patchFilter(newFilter);
      }),
      switchMap(() => service.filter$),
      take(1),
      tap(filters => {
        // Filter has been updated
        expect(filters).toMatchSnapshot();
      }),
    )
    .subscribe(() => {
      done();
    });
};

export const getByKeyFromApiTestLogic = (
  service: any,
  dataFixture: any,
  done: jest.DoneCallback,
) => {
  const sdkSpy = jest
    .spyOn(service.apiConf, 'get')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  service
    .getByKey$(dataFixture.id)
    .pipe(
      switchMap(() => service.entities$),
      take(1),
      tap(items => {
        // The entity list is empty
        expect(items).toMatchSnapshot();
      }),
      tap(() => {
        // "Get" method has been called
        expect((sdkSpy.mock.calls[0][0] as any).id).toMatchSnapshot();
      }),
      switchMap(() => service.entities$),
      take(1),
      tap(items => {
        // New item stored in the entity list
        expect(items).toMatchSnapshot();
      }),
    )
    .subscribe(() => {
      done();
    });
};

export const getByKeyFromCacheTestLogic = (
  service: any,
  dataFixture: any,
  done: jest.DoneCallback,
) => {
  jest
    .spyOn(service.apiConf, 'add')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  const sdkGetSpy = jest
    .spyOn(service.apiConf, 'get')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  service.entities$
    .pipe(
      take(1),
      tap(items => {
        // The entity list is empty
        expect(items).toMatchSnapshot();
      }),
      switchMap(() => service.add$(dataFixture)),
      tap(items => {
        // The new item has been added
        expect(items).toMatchSnapshot();
      }),
      delay(2000),
      switchMap(() => service.getByKey$(dataFixture.id)),
      tap(item => {
        // GET query has not been called
        expect(sdkGetSpy).not.toHaveBeenCalled();

        // Got the item from the cache
        expect(item).toMatchSnapshot();
      }),
    )
    .subscribe(() => {
      done();
    });
};

export const getByKeyFromCacheForceApiTestLogic = (
  service: any,
  dataFixture: any,
  done: jest.DoneCallback,
) => {
  jest
    .spyOn(service.apiConf, 'add')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  const sdkGetSpy = jest
    .spyOn(service.apiConf, 'get')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  service.entities$
    .pipe(
      take(1),
      tap(items => {
        // The entity list is empty
        expect(items).toMatchSnapshot();
      }),
      switchMap(() => service.add$(dataFixture)),
      tap(items => {
        // The new item has been added
        expect(items).toMatchSnapshot();
      }),
      delay(2000),
      switchMap(() => service.getByKey$(dataFixture.id, true, true)),
      tap(item => {
        // "Get" method has been called
        expect((sdkGetSpy.mock.calls[0][0] as any).id).toMatchSnapshot();

        // Got the item from the cache
        expect(item).toMatchSnapshot();
      }),
    )
    .subscribe(() => {
      done();
    });
};

export const getByKeyFromApiAndNotSaveToListTestLogic = (
  service: any,
  dataFixture: any,
  done: jest.DoneCallback,
) => {
  jest
    .spyOn(service.apiConf, 'add')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  const sdkGetSpy = jest
    .spyOn(service.apiConf, 'get')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  service.entities$
    .pipe(
      take(1),
      tap(items => {
        // The entity list is empty
        expect(items).toMatchSnapshot();
      }),
      switchMap(() => service.getByKey$(dataFixture.id, false, true)),
      tap(item => {
        // "Get" method has been called
        expect((sdkGetSpy.mock.calls[0][0] as any).id).toMatchSnapshot();

        // Got the item from the cache
        expect(item).toMatchSnapshot();
      }),
      switchMap(() => service.entities$),
      take(1),
      tap(items => {
        // New item NOT stored in the entity list
        expect(items).toMatchSnapshot();
      }),
    )
    .subscribe(() => {
      done();
    });
};
