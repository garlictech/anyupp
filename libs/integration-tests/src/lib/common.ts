import { devConfig as config } from '@bgap/shared/config';
export const testAdminUsername = 'test@anyupp.com';
export const testAdminUserPassword = 'Testtesttest12_';

const { GraphqlApiKey, GraphqlApiUrl } = config;
export { GraphqlApiUrl, GraphqlApiKey };
