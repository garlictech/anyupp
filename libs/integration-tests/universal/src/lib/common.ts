import { config } from '@bgap/shared/config';
export const testAdminUsername = 'test@anyupp.com';
export const testAdminUserPassword = 'Testtesttest12_';
import Amplify from 'aws-amplify';
import { awsConfig } from '@bgap/admin/amplify-api';

const { GraphqlApiKey, GraphqlApiUrl } = config;
export { GraphqlApiUrl, GraphqlApiKey };

export const configureAmplify = () => {
  Amplify.configure({
    ...awsConfig,
    // See: https://github.com/aws-amplify/amplify-js/issues/6552#issuecomment-682259256
    authenticationFlowType: 'USER_PASSWORD_AUTH',
    aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
  });
};
