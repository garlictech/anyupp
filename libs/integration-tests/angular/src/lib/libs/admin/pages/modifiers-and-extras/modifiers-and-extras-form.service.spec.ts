import { combineLatest, EMPTY, of } from 'rxjs';
import { catchError, switchMap, tap } from 'rxjs/operators';

import { HttpClientModule } from '@angular/common/http';
import { TestBed } from '@angular/core/testing';
import { FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
import { ModifiersAndExtrasFormService } from '@bgap/admin/pages/modifiers-and-extras';
import { entityConfig } from '@bgap/admin/shared/data-access/ngrx-data';
import { CrudSdkService } from '@bgap/admin/shared/data-access/sdk';
import { AdminSharedLoggedUserModule } from '@bgap/admin/store/logged-user';
import * as CrudApi from '@bgap/crud-gql/api';
import {
  productComponentSetFixture,
  testIdPrefix,
} from '@bgap/shared/fixtures';
import { UpsertResponse } from '@bgap/shared/types';
import {
  EntityCollectionServiceElementsFactory,
  EntityDataModule,
  EntityDispatcherFactory,
} from '@ngrx/data';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';

import { signInToCognito, signOutFromCognito } from '../../shared/helper';

describe('ModifiersAndExtrasFormService', () => {
  const productComponentId = `${testIdPrefix}ADMIN_PRODUCT_COMPONENT_IT_COMP_01`;
  const productComponentSetId = `${testIdPrefix}ADMIN_PRODUCT_COMPONENT_SET_IT_COMP_SET_01`;

  let service: ModifiersAndExtrasFormService;
  let crudSdk: CrudSdkService;

  const cleanup = () =>
    combineLatest([
      crudSdk.sdk.DeleteProductComponent({
        input: { id: productComponentId },
      }),
      crudSdk.sdk.DeleteProductComponentSet({
        input: { id: productComponentSetId },
      }),
    ]);

  beforeAll(async () => {
    await signInToCognito();
  }, 60000);

  afterAll(async () => {
    await signOutFromCognito();
  }, 60000);

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        ReactiveFormsModule,
        StoreModule.forRoot({}),
        EffectsModule.forRoot([]),
        EntityDataModule.forRoot(entityConfig),
        HttpClientModule,
        AdminSharedLoggedUserModule,
      ],
      providers: [
        EntityCollectionServiceElementsFactory,
        EntityDispatcherFactory,
      ],
    });

    service = TestBed.inject(ModifiersAndExtrasFormService);
    crudSdk = TestBed.inject(CrudSdkService);
  }, 60000);

  it('createProductComponentFormGroup should create form group', () => {
    expect(service.createProductComponentFormGroup().value).toMatchSnapshot();
  });

  it('addComponentToList/removeComponentFromList should handle component to item list', () => {
    const dialogForm = service.createProductComponentSetFormGroup();
    const componentForm = new FormGroup({
      productComponentId: new FormControl(''),
    });

    componentForm.patchValue({ productComponentId: 'productComponentId_01' });
    service.addComponentToList(dialogForm, componentForm);

    componentForm.patchValue({ productComponentId: 'productComponentId_02' });
    service.addComponentToList(dialogForm, componentForm);

    expect(dialogForm.value.items).toMatchInlineSnapshot(`
      Array [
        "productComponentId_01",
        "productComponentId_02",
      ]
    `);

    service.removeComponentFromList(dialogForm, 'productComponentId_01');

    expect(dialogForm.value.items).toMatchInlineSnapshot(`
      Array [
        "productComponentId_02",
      ]
    `);
  });

  it('saveComponentForm$ should call createProductComponent$ method when id is not specified', done => {
    const createSpy = jest
      .spyOn(service, 'createProductComponent$')
      .mockImplementationOnce(() => of({ data: 'ok', type: 'insert' }));

    service
      .saveComponentForm$(productComponentSetFixture.productComponentBase)
      .subscribe();

    expect(createSpy).toHaveBeenCalledWith(
      productComponentSetFixture.productComponentBase,
    );

    done();
  }, 60000);

  it('saveComponentForm$ should call updateProductComponent$ method when id is specified', done => {
    const updateSpy = jest
      .spyOn(service, 'updateProductComponent$')
      .mockImplementationOnce(() => of({ data: 'ok', type: 'update' }));

    service
      .saveComponentForm$(
        productComponentSetFixture.productComponentBase,
        productComponentId,
      )
      .subscribe();

    expect(updateSpy).toHaveBeenCalledWith({
      ...productComponentSetFixture.productComponentBase,
      id: productComponentId,
    });

    done();
  }, 60000);

  it('saveComponentForm$ should call updateProductComponent$ method with avoided dirty flag', done => {
    const updateSpy = jest
      .spyOn(service, 'updateProductComponent$')
      .mockImplementationOnce(() => of({ data: 'ok', type: 'update' }));

    service
      .saveComponentForm$(
        productComponentSetFixture.productComponentBase,
        productComponentId,
      )
      .subscribe();

    expect(updateSpy).toHaveBeenCalledWith({
      ...productComponentSetFixture.productComponentBase,
      id: productComponentId,
    });

    done();
  }, 60000);

  it('createProductComponent$ should create product component', done => {
    cleanup()
      .pipe(
        switchMap(() =>
          service.createProductComponent$({
            ...productComponentSetFixture.productComponentBase,
            id: productComponentId,
            chainId: 'chainId',
          }),
        ),
        tap(saveResponse => {
          expect(saveResponse).toMatchSnapshot({
            data: {
              createdAt: expect.any(String),
              updatedAt: expect.any(String),
            },
          });
        }),
        switchMap(() => cleanup()),
      )
      .subscribe(() => {
        done();
      });
  }, 60000);

  it('updateProductComponent$ should update product component', done => {
    cleanup()
      .pipe(
        switchMap(() =>
          service.createProductComponent$({
            ...productComponentSetFixture.productComponentBase,
            id: productComponentId,
            chainId: 'chainId',
          }),
        ),
        catchError(() => cleanup()),
        switchMap(saveResponse =>
          (<UpsertResponse<CrudApi.ProductComponent>>saveResponse).data.id
            ? service.updateProductComponent$({
                ...productComponentSetFixture.productComponentBase,
                id: productComponentId,
                chainId: 'chainId MOD',
              })
            : EMPTY,
        ),
        catchError(() => cleanup()),
        tap(updateResponse => {
          expect(updateResponse).toMatchSnapshot({
            data: {
              createdAt: expect.any(String),
              updatedAt: expect.any(String),
            },
          });
        }),
        switchMap(() => cleanup()),
      )
      .subscribe(() => {
        done();
      });
  }, 60000);

  it('saveComponentSetForm$ should call createProductComponentSet$ method when id is not specified', done => {
    const createSpy = jest
      .spyOn(service, 'createProductComponentSet$')
      .mockImplementationOnce(() => of({ data: 'ok', type: 'insert' }));

    service
      .saveComponentSetForm$(productComponentSetFixture.productComponentSetBase)
      .subscribe();

    expect(createSpy).toHaveBeenCalledWith(
      productComponentSetFixture.productComponentSetBase,
    );

    done();
  }, 60000);

  it('saveComponentSetForm$ should call updateProductComponentSet$ method when id is specified', done => {
    const updateSpy = jest
      .spyOn(service, 'updateProductComponentSet$')
      .mockImplementationOnce(() => of({ data: 'ok', type: 'update' }));

    service
      .saveComponentSetForm$(
        productComponentSetFixture.productComponentSetBase,
        productComponentSetId,
      )
      .subscribe();

    expect(updateSpy).toHaveBeenCalledWith({
      ...productComponentSetFixture.productComponentSetBase,
      id: productComponentSetId,
    });

    done();
  }, 60000);

  it('saveComponentSetForm$ should call updateProductComponentSet$ method with avoided dirty flag', done => {
    const updateSpy = jest
      .spyOn(service, 'updateProductComponentSet$')
      .mockImplementationOnce(() => of({ data: 'ok', type: 'update' }));

    service
      .saveComponentSetForm$(
        productComponentSetFixture.productComponentSetBase,
        productComponentSetId,
      )
      .subscribe();

    expect(updateSpy).toHaveBeenCalledWith({
      ...productComponentSetFixture.productComponentSetBase,
      id: productComponentSetId,
    });

    done();
  }, 60000);

  it('createProductComponentSet$ should create product component set', done => {
    cleanup()
      .pipe(
        switchMap(() =>
          service.createProductComponentSet$({
            ...productComponentSetFixture.productComponentSetBase,
            id: productComponentSetId,
            chainId: 'chainId',
          }),
        ),
        tap(saveResponse => {
          expect(saveResponse).toMatchSnapshot({
            data: {
              createdAt: expect.any(String),
              updatedAt: expect.any(String),
            },
          });
        }),
        switchMap(() => cleanup()),
      )
      .subscribe(() => {
        done();
      });
  }, 60000);

  it('updateProductComponentSet$ should update product component set', done => {
    cleanup()
      .pipe(
        switchMap(() =>
          service.createProductComponentSet$({
            ...productComponentSetFixture.productComponentSetBase,
            id: productComponentSetId,
            chainId: 'chainId',
          }),
        ),
        catchError(() => cleanup()),
        switchMap(saveResponse =>
          (<UpsertResponse<CrudApi.ProductComponentSet>>saveResponse).data.id
            ? service.updateProductComponentSet$({
                ...productComponentSetFixture.productComponentSetBase,
                id: productComponentSetId,
                chainId: 'chainId MOD',
              })
            : EMPTY,
        ),
        catchError(() => cleanup()),
        tap(updateResponse => {
          expect(updateResponse).toMatchSnapshot({
            data: {
              createdAt: expect.any(String),
              updatedAt: expect.any(String),
            },
          });
        }),
        switchMap(() => cleanup()),
      )
      .subscribe(() => {
        done();
      });
  }, 60000);
});
