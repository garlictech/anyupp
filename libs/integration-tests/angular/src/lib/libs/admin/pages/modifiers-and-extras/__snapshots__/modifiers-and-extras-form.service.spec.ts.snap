// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModifiersAndExtrasFormService createProductComponent$ should create product component 1`] = `
Object {
  "data": Object {
    "allergens": Array [
      "egg",
      "gluten",
      "milk",
      "soya",
      "peanut",
    ],
    "chainId": "chainId",
    "createdAt": Any<String>,
    "description": "DESCRIPTION",
    "dirty": null,
    "externalId": null,
    "id": "test_ADMIN_PRODUCT_COMPONENT_IT_COMP_01",
    "name": Object {
      "de": "PRODUCT_COMPONENT",
      "en": "PRODUCT_COMPONENT",
      "hu": "TERMEK_KOMPONENS",
    },
    "soldOut": null,
    "updatedAt": Any<String>,
  },
  "type": "insert",
}
`;

exports[`ModifiersAndExtrasFormService createProductComponentFormGroup should create form group 1`] = `
Object {
  "allergens": Array [],
  "chainId": "",
  "description": "",
  "name": Object {
    "de": "",
    "en": "",
    "hu": "",
  },
  "soldOut": false,
}
`;

exports[`ModifiersAndExtrasFormService createProductComponentSet$ should create product component set 1`] = `
Object {
  "data": Object {
    "chainId": "chainId",
    "createdAt": Any<String>,
    "description": "DESCRIPTION",
    "dirty": null,
    "externalId": null,
    "id": "test_ADMIN_PRODUCT_COMPONENT_SET_IT_COMP_SET_01",
    "items": Array [],
    "maxSelection": 0,
    "name": Object {
      "de": "COMPONENT_SET",
      "en": "COMPONENT_SET",
      "hu": "KOMPONENS_SET",
    },
    "supportedServingModes": null,
    "type": "extras",
    "updatedAt": Any<String>,
  },
  "type": "insert",
}
`;

exports[`ModifiersAndExtrasFormService updateProductComponent$ should update product component 1`] = `
Object {
  "data": Object {
    "allergens": Array [
      "egg",
      "gluten",
      "milk",
      "soya",
      "peanut",
    ],
    "chainId": "chainId MOD",
    "createdAt": Any<String>,
    "description": "DESCRIPTION",
    "dirty": null,
    "externalId": null,
    "id": "test_ADMIN_PRODUCT_COMPONENT_IT_COMP_01",
    "name": Object {
      "de": "PRODUCT_COMPONENT",
      "en": "PRODUCT_COMPONENT",
      "hu": "TERMEK_KOMPONENS",
    },
    "soldOut": null,
    "updatedAt": Any<String>,
  },
  "type": "update",
}
`;

exports[`ModifiersAndExtrasFormService updateProductComponentSet$ should update product component set 1`] = `
Object {
  "data": Object {
    "chainId": "chainId MOD",
    "createdAt": Any<String>,
    "description": "DESCRIPTION",
    "dirty": null,
    "externalId": null,
    "id": "test_ADMIN_PRODUCT_COMPONENT_SET_IT_COMP_SET_01",
    "items": Array [],
    "maxSelection": 0,
    "name": Object {
      "de": "COMPONENT_SET",
      "en": "COMPONENT_SET",
      "hu": "KOMPONENS_SET",
    },
    "supportedServingModes": null,
    "type": "extras",
    "updatedAt": Any<String>,
  },
  "type": "update",
}
`;
